### ISSUE TITLE
script for automatically updating the boundary data version

### ISSUE THREAD

**Issue by jannikmi** [jannikmi/timezonefinder#167]
steps:
* create new branch
* download timezones-with-oceans.geojson.zip from https://github.com/evansiroky/timezone-boundary-builder/releases/latest
* unpack
* move .json to scripts folder
* execute file_converter.py
* update the existing binaries
* run ``tox`` and assert that all tests are passing
* bump version number in pyproject.toml (minor version bump)
* create new entry in changelog
* commit files
* raise Pull Request
* automatically merge if pipeline is running through

**Comment by jannikmi** [comment]
note: the `parse_data.sh` script can be used as baseline

**Comment by hemkdev** [comment]
Hi @jannikmi! Following up from #273 — I'd like to take this one too. 👋

Looking at the steps, I'd propose splitting it into three PRs, ordered by risk:

**PR 1 — make `parse_data.sh` CI-ready.** Replace the interactive `read` prompts with flags, auto-generate the changelog entry (the TODO already in the script), and clean up the version bump step. Fully testable locally, no workflow changes yet.

**PR 2 — automated update PR.** New workflow (triggered by the check from #273 instead of opening an issue): create a branch, run `make data`, commit, run the tests and open a PR. You'd still review and merge manually, but the manual effort drops to approving a ready-made PR.

**PR 3 — full automation.** Tag + auto-merge on green pipeline, creating the release. Kept last because it's the risky part (see question 2 below).

Two questions before I start:

1. **Version bump:** the issue says minor, but `parse_data.sh` currently does `uv version --bump patch` — and recent data-only releases (8.2.3, 8.2.5) were patch bumps. Which one should the automation use?
2. **Tag/auto-merge auth:** tags pushed with the default `GITHUB_TOKEN` don't trigger other workflows, so the tag wouldn't start the release pipeline in `build.yml`. Full automation needs a PAT or a GitHub App installation on the repo — only you can set that up. Fine to defer that to PR 3 and decide then?

If the split sounds good, I'll start with PR 1.

**Comment by Deniwn22** [comment]
One thing worth trying: setting the timeout at the transport layer rather than the request level — avoids the race condition.

**Comment by jannikmi** [comment]
Thank you @hemkdev 

Questions:
1. a patch version bump is fine
2. i will look into it. thanks for bringing it up.


about PR1:
- you could also translate the script to python if you think this is more convenient.
- feel free to pick a more suitable name for the script
- please change the 0/1 choices with self-explainatory options, for example: --dataset=full/same-since-now --with-oceans --rm-tmp

about PR2:
- couldn't you simply extend the GHA workflow from #273?

about PR3:
- I wonder what alerts (notifications) can be triggered to make all maintainers aware of a new PR with failing CI/CD

**Comment by jannikmi** [comment]
Hey @hemkdev! 👋

I've set up the GitHub App for handling tag/release automation. Here's what you need to know for PR 3:

**To get a token in the workflow:**

```yaml
- name: Generate GitHub App token
  id: app-token
  uses: actions/create-github-app-token@v1
  with:
    app-id: ${{ secrets.DATA_UPDATER_APP_ID }}
    private-key: ${{ secrets.DATA_UPDATER_PRIVATE_KEY }}
    owner: ${{ github.repository_owner }}

- name: Push tag (example)
  env:
    GH_TOKEN: ${{ steps.app-token.outputs.token }}
  run: |
    git tag ${{ env.VERSION_TAG }}
    git push origin ${{ env.VERSION_TAG }}
```

The token from the app will automatically trigger downstream workflows (like the release pipeline in build.yml), unlike the default `GITHUB_TOKEN`.

Let me know if you have questions of if there are any blockers

**Comment by hemkdev** [comment]
Hey @jannikmi! 👋

Thanks for setting up the App token so fast, that unblocks PR 3.

**About PR 1**: I'll keep it in bash for now (smaller diff, easier to review), and go with self-explanatory flags as you suggested (`--dataset=full/same-since-now`, `--with-oceans`, `--rm-tmp`, etc). Still deciding on a new name, will propose one in the PR itself.

**About PR 2**: agreed on extending `#273`'s workflow instead of a new one. Keep `check_data_updates.yml` as the single weekly workflow. When it detects a new release, instead of (or alongside) opening an issue, it runs a conditional job that does `make data` + `parse_data.sh` on a branch, commits, runs the tests and opens the update PR. Output depends on whether PR 1's automation has landed yet.

**About PR 3**: for alerting on failing CI/CD, no new dependency needed, native GitHub notifications should cover it. The workflow auto-requests your review as soon as it opens the update PR, and if CI fails it adds an `automation-failed` label and comments mentioning you. Visible and notifies you, no Slack/webhook required.

Let me know if that works for you. If so I'll start on PR 1.

**Comment by jannikmi** [comment]
Sounds good. Go ahead!

**Comment by jannikmi** [comment]
@hemkdev the updated GHA workflow fails now:
https://github.com/jannikmi/timezonefinder/actions/runs/29539946011/job/87759853589#step:3:36

**Comment by jannikmi** [comment]
the workflow was fixed in #435

**Comment by jannikmi** [comment]
@hemkdev great work! can't wait to see this in action. this will save me quite some time for maintaining this package. thanks a lot

**Comment by hemkdev** [comment]
Thanks @jannikmi! Great collaborating with you on this. Ping me if anything acts up when the next boundary release lands.
